documentation/docs/guides/testing.md

```diff
@@ -0,0 +1,148 @@
+# Testing Fuego Controllers
+
+Fuego provides a `MockContext` type that makes it easy to test your controllers without using httptest, allowing you to focus on your business logic instead of the HTTP layer.
+
+## Using MockContext
+
+The `MockContext` type implements the `ContextWithBody` interface. Here's a simple example:
+
+```go
+func TestMyController(t *testing.T) {
+    // Create a new mock context with your request body type
+    ctx := fuego.NewMockContext[MyRequestType]()
+
+    // Set the request body
+    ctx.SetBody(MyRequestType{
+        Name: "John",
+        Age:  30,
+    })
+
+    // Call your controller
+    response, err := MyController(ctx)
+
+    // Assert the results, using the well-known testify library
+    assert.NoError(t, err)
+    assert.Equal(t, expectedResponse, response)
+
+    // Or, using the standard library
+    if err != nil {
+        t.Fatalf("unexpected error: %v", err)
+    }
+    if !reflect.DeepEqual(expectedResponse, response) {
+        t.Fatalf("unexpected response: %v", response)
+    }
```

Comment:
I don't think there is a need to document the alternative ways to test the error.

It brings confusion

Reply:
Yeah sorry, I'm the one who added it